libmate-desktop/mate-desktop.h

```diff
@@ -40,6 +40,7 @@ G_BEGIN_DECLS
         (MATE_MAJOR == (major) && MATE_MINOR == (minor) && \
         MATE_MICRO >= (micro)))
 
+#define MATE_DESKTOP_VERSION PACKAGE_VERSION
```

Comment:
How would that work?  This relies on `config.h` defining `PACKAGE_VERSION`, which is the *built* project's version.  So inside libmate-desktop's build it'll be OK, but outside it won't.

Admittedly, existing `MATE_DESKTOP_CHECK_VERSION` isn't gonna work either outside of libmate-desktop, but there's no reason to add more non-working things :smiling_face_with_tear:

Reply:
My negligence, ```MATE_DESKTOP_CHECK_VERSION``` should be removed

Reply:
Sure can do, but why do we need this?

Reply:
Provide a version interface for other external applications to obtain the ```mate-desktop``` version, such as the ```gdbus``` version interface of ```gnome-shell```  and the ```GNOME_DESKTOP_PLATFORM_VERSION``` macro provided by ```gnome -desktop```macro

Reply:
@zhuyaliang the autotools part is added.

For the rationale I'm not sure I understand the use case, but that's fine anyway, if you think it's useful, let's do it.
Should we make the `MATE_DESKTOP_CHECK_VERSION` work, or we still remove it?

Reply:
I searched for the code and didn't find the use of  ```MATE_DESKTOP_CHECK_VERSION```,If you can add support in ```Makefile.am```, I can add support in ```meson```